### PR TITLE
add support for event log API

### PR DESCRIPTION
**Pull request by anneFly**
This PR adds support for the [event log API](https://developer.close.io/#event-log).
The testing stub will not record event logs by default but I added a context manager that can be used to record logs while testing so that the `get_event_logs` method of the testing stub will return useful data. However, for now only task creation and deletion is supported as possible event logs.

### CODE REVIEW

File: closeio/contrib/testing_stub.py [hunk 1]
```diff
@@ -2,13 +2,85 @@
...
+    It's best to use it only for single operations and NOT call
+    multiple operations inside the context manager so that the
+    order of event logs will remain correct.
+
+    Currently only event log data for task creation and deletion
+    is supported.
+
+    Usage::
+        with record_logs(stub):
+            task = stub.create_task(lead_id='x', assigned_to='y', text='z')
```
Comment (anapaulagomes) [inline closeio/contrib/testing_stub.py]: 👍

File: closeio/utils.py [hunk 2]
```diff
@@ -160,6 +160,32 @@ def paginate(func, *args, **kwargs):
             skip += limit
 
 
+def paginate_via_cursor(func, *args, **kwargs):
```
Comment (anapaulagomes) [inline closeio/utils.py]: Shall we add a test for it?

Comment (anneFly) [reply closeio/utils.py]: hmm... yes you're right, a test would be nice. I need to think about how I can test this properly without creating hundreds of event logs in the test. I'll think about it and will not merge it until then.

File: closeio/contrib/testing_stub.py [hunk 3]
```diff
@@ -36,6 +41,54 @@ def _data(self, attr, default=None):
...
+        is supported.
+
+        Example usage::
+            with stub.record_logs():
+                task = stub.create_task(lead_id='x', assigned_to='y', text='z')
+                stub.delete_task(task_id=task['id'])
+        """
+        self.record_event_logs = True
+        yield
+        self.record_event_logs = False
```
Comment (anapaulagomes) [inline closeio/contrib/testing_stub.py]: You're right, this way looks nicer and easier to understand. Cool. :)

File: closeio/contrib/testing_stub.py [hunk 4]
```diff
@@ -36,6 +41,54 @@ def _data(self, attr, default=None):
...
+        log = self._create_task_log(task)
+        log.update({
+            'action': 'deleted',
+            'data': {},
+            'previous_data': task,
+        })
+        logs.insert(0, log)
+
+    @contextmanager
+    def record_logs(self):
```
Comment (anapaulagomes) [inline closeio/contrib/testing_stub.py]: Sorry for coming back to this topic again but I was thinking about the bug you caught yesterday. Looking at the usage example of `record_logs`, I think that would be nice adding a test for it too. Something like:

```
    def test_record_logs(self):
        task = stub.create_task(lead_id='x', assigned_to='y', text='z')
        stub.delete_task(task_id=task['id'])
        
        logs = client.get_event_logs(object_type='task.lead')
        logs = list(logs)
        assert len(logs) == 0
        
        with stub.record_logs():
            task = stub.create_task(lead_id='x', assigned_to='y', text='z')
            stub.delete_task(task_id=task['id'])
        
        logs = client.get_event_logs(object_type='task.lead')
        logs = list(logs)
        assert len(logs) == 2
```

What do you think?

**Comment by anapaulagomes** [comment]
Good to go! 👍

**Pull request merged**